README.md

```diff
@@ -159,33 +159,131 @@ We request that our customers adhere to the following guidelines to avoid
-page](https://github.com/googleapis/google-cloud-cpp/releases).
+## Beyond the C++ API
 
-Note that this document has no bearing on the Google Cloud Platform deprecation
-policy described at https://cloud.google.com/terms.
+Applications developers interact with a C++ library through more than just
+the C++ symbols and headers. They also need to reference the name of the
+library in their build scripts. Depending of the build system they use
+this may be a CMake target, a Bazel rule, a pkg-config module, or just the
+name of some object in the file system.
+
+As with the C++ API, we try to avoid breaking changes to these interface
+points. Sometimes such changes yield benefits to our customers, in the form of
+easier-to-understand what names go with with services, or more consistency
+across services. When these benefits warrant it, we will announce these changes
+prominently in our `CHANGELOG.md` file and in the affected release's notes.
+Nevertheless, though we take commercially reasonable efforts to prevent this,
+it is possible that backwards incompatible changes go undetected and,
+therefore, undocumented. We apologize if this is the case and welcome feedback
+or bug reports to rectify the problem.
+
+### Experimental Libraries
+
+From time to time we add libraries to `google-cloud-cpp` to validate new
+designs, expose experimental (or otherwise not generally available) GCP
+features, or simply because a library is not yet complete. Such libraries
+will have `experimental` in their CMake target and Bazel rule. The README
+file for these libraries will also document that they are experimental.
+Such libraries are subject to change, including removal, without notice.
+This includes, but it is not limited to, all their symbols, pre-processor
+macros, files, targets, rules, and installed artifacts.
+
+### Bazel rules
+
+Only the rules exported at the top-level directory, except experimental rules,
+are part of the public API (e.g. `//:spanner`). Experimental rules have
+`experimental` in their name, e.g. `//:experimental-firestore`).
+
+Previously we some of the rules in subdirectories
+(e.g. `//google/cloud/bigtable:bigtable_client`) had public visibility. These
+rules are deprecated as of 2021-02-15, and will be become inaccessible
+(or removed) on or shortly after **2022-02-15**.
+
+### CMake targets and packages
+
+Only CMake packages starting with the `google_cloud_cpp_` prefix are part of
+the public API. Only targets starting with `google-cloud-cpp::`, except
+experimental targets, are part of the public API. Experimental targets have
+`experimental` in their name (e.g. `google-cloud-cpp::experimental-iam`).
+
+In previous versions we released packages with other prefixes (or without
+specific prefixes), these are deprecated as of 2021-02-15, and will be retired
+on or shortly after **2022-02-15**. Same applies to any targets exported with
+other prefixes (or without an specific prefix).
+
+#### CMake targets outside packages
```

Comment:
Do we need this section? Maybe I don't really understand what this section is saying or who it might be relevant to.

Reply:
The problem is customers can (and have, maybe still do!) use the library by making `google-cloud-cpp` a submodule, or using CMake's
[FetchContent](https://cmake.org/cmake/help/latest/module/FetchContent.html).  In that case *all* our targets are visible to the application, because (a) both these techniques use `add_subdirectory()` to add the project into the customer's CMake build, and (b) target names are global.